flytekit/tools/translator.py

```diff
@@ -163,6 +191,64 @@ def fn(settings: SerializationSettings) -> List[str]:
     return fn
 
 
+def _update_serialization_settings_for_ipython(
+    entity: FlyteLocalEntity,
+    serialization_settings: SerializationSettings,
+    options: Optional[Options] = None,
+) -> SerializationSettings:
+    # If the entity is not a PythonAutoContainerTask, we don't need to do anything, as only Tasks with container |
+    # user code in container need to be serialized as pickled objects.
+    if not isinstance(entity, (PythonAutoContainerTask, ArrayNodeMapTask)):
+        return serialization_settings
+
+    from flytekit.tools.interactive import ipython_check
+
+    # Let's check if we are in an interactive environment like Jupyter notebook
+    if ipython_check():
```

Comment:
This is what I was referring to, if instead of this we can check the settings from a Context.
the context could be explicitly set or we could set it based on ipython availability